src/components/Form/index.tsx

```diff
@@ -0,0 +1,171 @@
+import * as React from 'react';
+
+import type {
+    Comment,
+    CommentReply,
+} from '../../state/comments';
+
+
+export interface CommentFormSetProps {
+    comments: Comment[];
+}
+
+export function CommentFormSetComponent(props: CommentFormSetProps) {
+    const [initialNumber] = React.useState(props.comments.filter(comment => {return comment.remoteId != null}).length);
+    const prefix = "comments";
+
+    const commentForms = props.comments.map((comment, formNumber) => (
+        <CommentFormComponent
+            key={comment.localId}
+            comment={comment}
+            formNumber={formNumber}
+        />
+    ));
+
+    return (<>
+        <PrefixedHiddenInput
+            fieldName="TOTAL_FORMS"
+            value={props.comments.length}
+            prefix={prefix}
+        />
+        <PrefixedHiddenInput
+            fieldName="INITIAL_FORMS"
+            value={initialNumber}
+            prefix={prefix}
+        />
+        <PrefixedHiddenInput
+            fieldName="MIN_NUM_FORMS"
+            value="0"
+            prefix={prefix}
+        />
+        <PrefixedHiddenInput
+            fieldName="MAX_NUM_FORMS"
+            value=""
+            prefix={prefix}
+        />
+        {commentForms}
+    </>)
+}
+
+interface PrefixedHiddenInputProps {
+    prefix: string;
+    value: number|string;
+    fieldName: string;
+}
+
+function PrefixedHiddenInput(props: PrefixedHiddenInputProps) {
+    return (<input
+        type="hidden"
+        name={`${props.prefix}-${props.fieldName}`}
+        value={props.value} 
```

Comment:
**Question:** How did you get this to compile? I did not know you can pass number straight to `<input>` value.

Reply:
Honestly I just didn't have any problems with this, it just compiled with no complaints.

Reply:
https://github.com/microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L7296 mhm.... weird!